app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,34 @@
 package com.lightricks.feedexercise.data
 
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.GetFeedResponse
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
 class FeedRepository {
-    //todo: implement
-}
\ No newline at end of file
+
+    private val baseURI: String = "https://assets.swishvideoapp.com/"
+
+    suspend fun getTemplatesMetadataItemsList(): GetFeedResponse {
+        val moshi: Moshi = Moshi.Builder()
+            .add(KotlinJsonAdapterFactory())
```

Comment:
We don't really need this adapter if we are generating adapter for our DTO. So we can remove it.

Reply:
the app crashes without this line

Reply:
What exception did you get? Because we shouldn't.

Reply:
java.lang.RuntimeException: Failed to find the generated JsonAdapter class for class com.lightricks.feedexercise.network.GetFeedResponse

Reply:
That is because we are not applying the annotation processor, so the adapters are not generated. 
https://github.com/square/moshi#codegen

We have to add kapt("com.squareup.moshi:moshi-kotlin-codegen:1.14.0") as a dependency.